src/diffusers/configuration_utils.py

```diff
@@ -292,9 +302,36 @@ def extract_init_dict(cls, config_dict, **kwargs):
             for arg in cls._flax_internal_args:
                 expected_keys.remove(arg)
 
+        # 2. Remove attributes that cannot be expected from expected config attributes
         # remove keys to be ignored
         if len(cls.ignore_for_config) > 0:
             expected_keys = expected_keys - set(cls.ignore_for_config)
+
+        # load diffusers library to import compatible and original scheduler
+        diffusers_library = importlib.import_module(__name__.split(".")[0])
+
+        # remove attributes from compatible classes that orig cannot expect
+        compatible_classes = [getattr(diffusers_library, c, None) for c in cls._compatible_classes]
+        # filder out None potentially undefined dummy classes
+        compatible_classes = [c for c in compatible_classes if c is not None]
+        expected_keys_comp_cls = set()
+        for c in compatible_classes:
+            expected_keys_c = cls._get_init_keys(c)
+            expected_keys_comp_cls = expected_keys_comp_cls.union(expected_keys_c)
+        expected_keys_comp_cls = expected_keys_comp_cls - cls._get_init_keys(cls)
```

Comment:
Does this mean that we accept all keys from all compatible classes? For example, let's say we have schedulers `A`, `B`, `C` and `C` defines a config var `c` that neither `A` nor `B` use. If we load `A` from a config saved by `C` then it is accepted. If we load `C` from a config saved by `A` we should still get a warning, right?

Reply:
We'll get an info statement then, will add a test for this :-)

Reply:
See here: https://github.com/huggingface/diffusers/pull/890/files#r1009617164 think that covers your use case no? 
IMO, we should not throw a warning but just an info statement because it just means that one parameter defaults